### PR TITLE
Dev/refactor e2e into po e2e

### PR DESCRIPTION
**Pull request by LedjoLleshaj**
Finished Refactoring the existing e2e tests to use the PageObject Pattern. Added Courses PO and CourseDetails PO with web elements and methods implemented.

### CODE REVIEW

**Comment by eliaperantoni** [comment]
Hey @LedjoLleshaj this is some great work! I just have one complaint: I see you use XPaths to target specific elements in the webpage but they seem to be excessively tied to the specific HTML structure instead of taking advantage of the semantics.

Take this for instance:
```java
@FindBy(xpath = "//body/div[1]/a")
private WebElement logout;
```
it may work now, but it wouldn't work anymore if we ever decided to move the button somewhere else in the page, add another link before the Logout, or wrap the login info box in a new `<div>` container, etc.. making the test quite fragile.

Something like:
```java
@FindBy(xpath = "//a[@href='/logout']")
private WebElement logout;
```
or even
```java
@FindBy(linkText = "Logout")
private WebElement logout;
```
I think would be much more resilient to changes in the HTML, and also better convey the semantic meaning of the button that we're trying to select. (These examples above _should_ work but I haven't tested them out. Hope my message is understandable still)

Do you think this is a valid criticism? If so, I'd suggest changing the page objects to use this style of XPaths that don't simply look at the current HTML structure but try to select the specific element in a more robust way. Feel free to add `id` to HTML tags to make them easier to select when there's no good alternative!

**Comment by LedjoLleshaj** [comment]
Yes i agree with you completely.This Refactoring was a little tricky since i was missing some page logic and the goal of this branch was to refactor the old e2e test and make the tests work for now PageObject weren't supposed to be part of this branch so i took the shortcut ,but yes what u proposed is a better way and will definitely fix them asap.

**Comment by LedjoLleshaj** [comment]
I believe now they are more flexible than before and resilient to future changes.

**Comment by eliaperantoni** [comment]
Awesome! They definitely look way better, I'll make a proper review ASAP.

File: src/main/resources/templates/course/list.html [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
+        <meta name="viewport"
+              content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+        <meta http-equiv="X-UA-Compatible" content="ie=edge">
+        <script src="https://cdn.tailwindcss.com?plugins=forms,typography,aspect-ratio,line-clamp"></script>
+        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.2.1/css/all.min.css" integrity="sha512-MV7K8+y+gLIBoVD59lQIYicR65iaqukzvf/nwasF0nqhPay5w/9lJmVM2hMDcnK1OnMGCdVK+iQrJ7lzPJQd1w==" crossorigin="anonymous" referrerpolicy="no-referrer" />
+        <link rel="stylesheet" href="/style.css">
+        <title>Ettore</title>
     </head>
     <body>
-        <!-- Header with logo, logout and breadcrumbs -->
```
Comment (nicofretti) [inline src/main/resources/templates/course/list.html]: Take a look at the use of commons components, actually We are using them on the branch develop

**Comment by eliaperantoni** [comment]
I've reverted the changes to `course` templates which came from `develop` with major modifications by @nicofretti. I'll merge this now but I want to clean up the project a bit i.e. by:
- Removing the unnecessary `/index` endpoint
- Translating everything to english
- Making all E2E tests pass
I'll do that today

**Pull request merged**